From 1567ca53cc8872e411ec15aad806e3f4958eace7 Mon Sep 17 00:00:00 2001
Message-ID: <1567ca53cc8872e411ec15aad806e3f4958eace7.1689663214.git.stefan@agner.ch>
In-Reply-To: <2e3653d9966a55355a8706b94ff47873677f6264.1689663214.git.stefan@agner.ch>
References: <2e3653d9966a55355a8706b94ff47873677f6264.1689663214.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 15 Jun 2023 04:02:06 +0200
Subject: [PATCH] [rest] deactive IPv6 interface when disabling the OTBR
 (#1904)

Deactivate the IPv6 interface when disabling the OTBR. This follows the
Attach/Deatach semantics of the D-Bus API.
---
 src/rest/resource.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/rest/resource.cpp b/src/rest/resource.cpp
index d8f2a39876..db8aaa41f1 100644
--- a/src/rest/resource.cpp
+++ b/src/rest/resource.cpp
@@ -359,13 +359,14 @@ void Resource::SetDataState(const Request &aRequest, Response &aResponse) const
     {
         if (!otIp6IsEnabled(mInstance))
         {
-            SuccessOrExit(otIp6SetEnabled(mInstance, true));
+            VerifyOrExit(otIp6SetEnabled(mInstance, true) == OT_ERROR_NONE, error = OTBR_ERROR_INVALID_STATE);
         }
         VerifyOrExit(otThreadSetEnabled(mInstance, true) == OT_ERROR_NONE, error = OTBR_ERROR_INVALID_STATE);
     }
     else if (body == "disable")
     {
         VerifyOrExit(otThreadSetEnabled(mInstance, false) == OT_ERROR_NONE, error = OTBR_ERROR_INVALID_STATE);
+        VerifyOrExit(otIp6SetEnabled(mInstance, false) == OT_ERROR_NONE, error = OTBR_ERROR_INVALID_STATE);
     }
     else
     {
-- 
2.41.0

